runtime-modules/content/src/lib.rs

```diff
@@ -1419,12 +1397,24 @@ impl<T: Trait> Module<T> {
         channel_id: &T::ChannelId,
         prize_source_account: &T::AccountId,
     ) -> DispatchResult {
-        // construct upload params
-        let upload_params =
-            Self::pick_upload_parameters_from_assets(assets, channel_id, prize_source_account);
+        // dynamic bag for the channel
+        let dyn_bag = DynamicBagIdType::<T::MemberId, T::ChannelId>::Channel(*channel_id);
+        let bag_id = BagIdType::from(dyn_bag.clone());
+
+        let upload_params = UploadParametersRecord {
+            bag_id: bag_id.clone(),
+            object_creation_list: assets.object_creation_list.clone(),
+            deletion_prize_source_account_id: prize_source_account.clone(),
+            expected_data_size_fee: assets.expected_data_size_fee,
+        };
+
+        // create dynamic bag if it doesn't exist (does not emits conflicting created/deleted bag events)
+        if let Err(_) = Storage::<T>::ensure_bag_exists(&bag_id) {
```

Comment:
Try this one: https://doc.rust-lang.org/std/result/enum.Result.html#method.is_err